### DISCUSSION TITLE
TWAIN: Doesn't detect scanner after NAPS2 Update to  7.3.1 or  7.3.0

### DISCUSSION THREAD

**Discussion by zsolks** [cyanfish/naps2#351] [Support]
I have many profiles with TWAIN config, and have used it form years. My scanner is printer Brother MFC7460DN with paper tray to scan.
After update NAPS2 to version 7.3.1 or  7.3.0 all profiles stopped working. When I tried modify profile can't find scanner and I have error as on attachment. TWAIN is unusable. Changing TWAIN version-implementation do not change anything. 
With WIA configuration I can scan but:
- when paper tray is empty i have to manually select as a source scanner glass
- when printer goes sleep it is impossible to wake it up and have to go to printer, push the button to NAPS start working (Before that i have message that scanner is busy).
Before update all this thing worked automatically.
Is there any solution to this problem. NAPS was great tool before this problems occurred.

![NAPS2](https://github.com/cyanfish/naps2/assets/155647580/c1d01fca-3a0b-4d75-ac5e-a9339aa1bb08)

**Comment by cyanfish** [comment]
Can you follow [these instructions](https://www.naps2.com/doc/troubleshooting#error-log) to collect error logs?

Also, is there any chance you've been using different installer types, e.g. used the .msi installer to upgrade when you previously used the .exe installer? If so, try uninstalling and reinstalling NAPS2.

**Comment by zsolks** [comment]
Problem solved.
For clearance, I'm using portable version on Windows. When there is new version I download zip, extract and overwrite old files.

What I done to solve problem:
I installed windows version, and this version worked. So I tested my portable version - didn't worked.
I Uninstaledl version from Windows. Portable version still not worked.
I downloaded new (new means fresh from website) portable version, unpack and start it. Fresh portable version was working with TWAIN.

So i compared my normally used portable version of NAPS2 with fresh version. Both version was 7.4.2.0
Fresh version had less files than my normally used NAPS2. There was more dll files. Probably this files was generating problem.

On the end, I putted in to fresh version of NAPS2 my config and profile. Everything work fine now.

Maybe it will help somebody with the same problem.

**Reply by cyanfish** [reply]
If you want to update the portable version, you should delete the old App folder and replace it with the new one. Or you can select "Check for updates" in the About menu and it'll do it for you.
